src/main/java/com/iota/iri/utils/log/interval/IntervalLogger.java

```diff
@@ -0,0 +1,490 @@
+                        outputThread.interrupt();
+                    }
+                }
+            }
+        } else {
+            scheduleOutput();
+        }
+    }
+
+    /**
+     * This method schedules the output of the last received message by spawning a {@link Thread} that will print the
+     * new message after the given timeout.
+     *
+     * When creating the {@link Thread} it copies its name so the output is "transparent" to the user and the effect
+     * is the same as dumping the message manually through the {@link org.slf4j.Logger} object itself.
+     */
+    private void scheduleOutput() {
+        if (outputThread == null) {
+            synchronized(this) {
+                if (outputThread == null) {
+                    outputThread = new Thread(() -> {
+                        try {
+                            Thread.sleep(Math.max(logInterval - (System.currentTimeMillis() - lastLogTime), 1));
+
+                            outputThread = null;
+
+                            triggerOutput(true);
+                        } catch(InterruptedException e) {
+                            outputThread = null;
+
+                            Thread.currentThread().interrupt();
+                        }
+                    }, Thread.currentThread().getName());
+
+                    outputThread.start();
+                }
+            }
+        }
+    }
+
+    /**
+     * Represents the basic logic that is shared by all message types that are supported by this logger.
+     */
+    private abstract class Message {
+        /**
+         * Holds the message that shall get printed.
+         */
+        protected final String message;
+
+        /**
+         * Creates a message that gets managed by this logger.
+         *
+         * It simply stores the provided message in the internal property.
+         *
+         * @param message message that shall get printed
+         */
+        public Message(String message) {
+            this.message = message;
+        }
+
+        /**
+         * This method handles the actual output of messages through the {@link org.slf4j.Logger} instance.
+         *
+         * It first checks if the message that shall get printed differs from the last message that was printed and then
+         * issues the output of the message. After printing the message, it updates the internal variables to handle the
+         * next message accordingly.
+         */
+        public abstract void print();
+
+        /**
+         * This method allows us to used the Messages in HashMaps and other collections that use the hashCode method to
+         * identify objects. Even tho this is not actively used we anyway want to implement it to be on the safe side
+         * if we ever implement something like this.
+         *
+         * @return a unique int identifier that can be used to address equal objects
+         */
+        @Override
+        public abstract int hashCode();
```

Comment:
Yes i want to force the subclasses to implement it.

Please also note that all of the classes implement a different version of this function.

result starts with a different number to account for the different "types" + an error has for example an additional property

Reply:
We could also use

    return Objects.hash(this.getClass(), message);

but it internally does the same as i do here manually but with the overhead of creating a new array internally for the passed in parameters - so i thought it would be faster to just write it manually.

Reply:
I see that you changed it and now there no subclasses?

Reply:
I think that even if you use the same implementation it won't be that bad because:
1. Hash collision would rarely occur because the message will be different
2. Even if a collision would occur then it isn't that bad. A collision is allowed to happen. There will simply be 2 different messages in the same bucket of the HashMap
3. Currently you don't even put them in a HashMap and I doubt that you will. And even if you will it won't matter because of the other points.

I suppose you are correct about `Objects.hash`...
I am not sure if it is a reason to totally dismiss it though

Reply:
For error you can call `super` and then apply your logic